packages/eslint-plugin/tests/rules/ban-ts-comment.test.ts

```diff
@@ -34,6 +34,17 @@ ruleTester.run('ts-expect-error', rule, {
         },
       ],
     },
+    {
+      code: '// @ts-expect-error: TS1234 because xyz',
+      options: [
+        {
+          'ts-expect-error': {
+            descriptionFormat: '^: TS\\d+ because .+$',
```

Comment:
Honestly I don't mind this API at all. Regular expressions are pretty ideal for this scenario. Maybe `description` would be a cleaner, more succinct name?

Here's a fun one though: what if people want to enforce the specific error message(s) getting reported, as in https://github.com/microsoft/TypeScript/issues/19139? Do you see a clean way to add those to the regex?

Reply:
When I said I don't like this API, I was more saying the union of "boolean + string + object", which looks all a bit ad-hoc.

As for `description`, my main concern is we would have more description-related config in the future—e.g. maybe we want to configure `descriptionLength` per directive?

> what if people want to enforce the specific error message(s) getting reported

No without making it a huge, huge regex union😅 I mean, at rate it would mean including the entire `diagnosticMessages.json` in the rule, whether as config or as part of the rule itself, right

Reply:
Oh I mean, what if the user wants to verify that the TS* error mentioned in the comment is the same as what TypeScript is reporting?

For this example, some might want to have the rule say that the `TS0000` should actually be `TS2322` [[playground](https://www.typescriptlang.org/play?#code/FDDGHsDsGcBcAIBuBDANgVwKYC56XQLYBGmATvALzwDkyRo1A3EA)]:

```ts
// @ts-expect-error -- TS0000: this is fine
const value: number = 'abc';
```

Reply:
Right, we can only verify that by actually reading the entire `diagnosticMessages.json`. The user can do that as well by reading `diagnosticMessages.json` and building a regex union.

Reply:
OH I think I see what you're saying - do you mean verifying that the `TS0000` message matches the rest of the _comment_? In that example, `this is fine`?

I'm suggesting asking the _TypeScript compiler_ what the message should be. But looking back though the compiler APIs I don't see anything particularly useful.

Anyway, if we end up finding some TS API like `.getDiagnosticsDespiteSuppression()`, I suppose we could add a named capture group for it. So regular expressions seems fine to me!

Reply:
Ah, I get what you mean, you were suggesting using something _other_ than a user-defined regex. Yeah that would mean consulting some kind of TS API.

But it will be even more confusing until TS actually implements per-error suppression...